### DISCUSSION TITLE
Dynamically generated NAV menu

### DISCUSSION THREAD

**Discussion by lminuti** [vuejs/vitepress#4404] [Q&A] [answered]
I need to create some Nav menu items from an ajax call. That ajax call should be done at runtime: when someone visit the web site, I have to read some data from a remote API and build some menu items.  Have you some idea?

**Comment by brc-dd** [comment, marked as answer]
https://vitepress.dev/reference/default-theme-nav#custom-components

**Reply by lminuti** [reply]
Thank you. I think it's exactly what I need.

**Reply by brc-dd** [reply]
You can refer these components - https://github.com/vuejs/vitepress/tree/main/__tests__/e2e/.vitepress/theme/components

If you're not familiar with vue, the fetch will be something like this:

```ts
import { onMounted, ref } from 'vue'

const props = defineProps<{
  screenMenu?: boolean
}>()

const items = ref<{ text: string; link: string }[]>([])

onMounted(async () => {
  const res = await fetch('https://your/api/call')
  const json = await res.json()
  // process json 
  items.value = processedJson
})
```
